### DISCUSSION TITLE
Is there a limit on the amount of times you can sign in?

### DISCUSSION THREAD

**Discussion by tdunnigan** [mmohades/Venmo#24] [Q&A] [answered]
I've been building an application that uses your venmo wrapper.  First, thank you!  It is awesome.  But I've been signing into my own account over and over to test the different functions, and suddenly I am receiving the error below and I cannot sign in.  

venmo_api.models.exception.HttpCodeError: HTTP Status code is invalid. Could not make the request because -> 400 Bad Request.
Error: {'error': {'message': 'Rate limit exceeded.', 'code': 277, 'links': None, 'title': 'Error'}}

Is this a daily limit of some sort?  Hoping this doesn't mean I can't sign in again with my account!

Thank you in advance!

-Tommy

**Comment by mmohades** [comment, marked as answer]
Hey Tommy,

From my experience, there's a ~120 request rate limit for all the endpoints together. After an hour, you'll able to make requests again. However, if this keeps happening, like let's say for a day, Venmo might block your account/IP address. I'm not sure what would be the next step to resolve it.
I'd suggest trying to minimize your request. For example, you only need to sign in once. Once you get your access token, you can use it until you revoke it manually using `client.log_out('token goes here')`
I believe there's also the error code `429` that is related to reaching the rate limit.

**Comment by tdunnigan** [comment]
Thank you for your quick reply!  I appreciate it.  I'll limit my requests
and hopefully avoid this problem in the future.

Happy Holidays!

Best,
Tommy

On Wed, Dec 23, 2020 at 5:20 PM Mark Mohades <notifications@github.com>
wrote:

> Hey Tommy,
>
> From my experience, there's a ~120 requests rate limit for all the
> endpoints together. After an hour, you'll able to make requests again.
> However, if this keep happening, like let's say for a day, Venmo might
> block your account/IP address. I'm not sure what would be the next step to
> resolve it.
> I'd suggest try to minimize your request. For example, you only need to
> sign in once. Once you get your access token, you can use it as until you
> revoke it manually using client.log_out('toke goes here')
> I believe there's also the error code 429 that is related to reaching the
> rate limit.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mmohades/Venmo/discussions/24#discussioncomment-238836>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQDDRVU3PP7J5AHSZNYKIM3SWKJNLANCNFSM4VHUFVEQ>
> .
>


-- 
Tommy Dunnigan
tdunni88@gmail.com
916-879-0024

**Reply by mmohades** [reply]
No problem. If what you're doing doesn't involve money transfer (send/request money), then creating another account only for what you're doing can be helpful as well. I hope that helps. Happy Holidays!
